Dockerfile

```diff
@@ -0,0 +1,32 @@
+# Dockerfile
+FROM barichello/godot-ci:4.3
+
+# Set working directory
+WORKDIR /app
+
+RUN apt update -y && apt install -y scons g++
+RUN wget https://github.com/godotengine/godot-cpp/archive/refs/tags/godot-4.3-stable.tar.gz
+RUN tar -xf godot-4.3-stable.tar.gz
+RUN mv godot-cpp-godot-4.3-stable godot-cpp
+
+RUN wget https://github.com/bitwes/Gut/archive/refs/tags/v9.3.0.tar.gz
+RUN tar -xf v9.3.0.tar.gz
+RUN mkdir vendor && mv Gut-9.3.0 vendor/gut
+
+COPY SConstruct ./SConstruct
+RUN scons
+
+# Copy project files
+COPY addons ./addons
+COPY src ./src
+RUN mkdir -p demo/addons demo/bin
+RUN scons
+
```

Comment:
Scons runned twice without need for that

Reply:
Must run twice. Every docker build command creates a new layer (overrlayfs), which can be cached. First run is compiling godot-cpp. The second is compiling our code and linking them together, saving the result in a new layer.

Reply:
I think that you can use cache from scons builds for Godot cpp there tho, so docker wont need to build it at all